src/CharacterList.test.js

```diff
@@ -0,0 +1,40 @@
+import React from 'react';
+import { shallow } from 'enzyme';
+import CharacterList from './CharacterList';
+
+it('renders without crashing', () => {
+  shallow(<CharacterList params={{}} />);
+});
+
+it('Increments page when clicked', () => {
+  const component = shallow(
+    <CharacterList params={{}} />
+  );
+
+  expect(component.state('page')).toEqual(0);
+
+  component.setState({
```

Comment:
by setting state to create the button, you have some logic in your test that you are kind of implicitly testing for. that creates some confusion for me in reading the test, and can lead to some difficulty in trying to use the test later on to troubleshoot any potential bug that you're having. 

it may be beneficial to first test if the button is there when you set the state as you have. then, you could test whether clicking that button performs the action that you think it should. 

This may be a case for breaking that button into it's own component to make it easier to test, although it might still be complex because of the way it updates state.